enterprise/coderd/roles_test.go

```diff
@@ -0,0 +1,63 @@
+package coderd_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/coderdtest"
+	"github.com/coder/coder/v2/coderd/database/db2sdk"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/enterprise/coderd/coderdenttest"
+	"github.com/coder/coder/v2/enterprise/coderd/license"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestCustomRole(t *testing.T) {
```

Comment:
What happens if we have existing custom roles, our license expires, and we try to add more / modify?

What happens if we omit certain fields from the payload? Do the unspecified entries get overwritten in the database?

Reply:
Roles are still usable, just not editable after the license expires. Fields all override. 

I'll add more tests. Most of my tests are in the `dbauthz` layer for the actual escalation and security related assertions.

Reply:
Added